tools/discovery/builder.rb

```diff
@@ -0,0 +1,138 @@
+$LOAD_PATH.unshift File.join(File.dirname(__FILE__), '..', '..')
+require 'net/http'
+require 'json'
+require 'erb'
+require 'compile/core'
+
+# rubocop:disable Style/MixinUsage
+include Compile::Core
+# rubocop:enable Style/MixinUsage
+
+
+# Represents a property on a resource
+class Property
+  attr_reader :name
+  def initialize(name, attributes, json)
+    @name = name
+    @attributes = attributes
+    @json = json
+  end
+
+  def type
+    return 'NestedObject' if @attributes['$ref']
+    @attributes['type'].capitalize
+  rescue StandardError
+    'Not yet Implemented'
+  end
+
+  def description
+    @attributes['description']
```

Comment:
I wouldn't mind seeing the [Output-only]s stripped from the description

Reply:
Agreed! Still an ongoing TODO at the moment.